tool/tsh/common/db.go

```diff
@@ -975,24 +966,27 @@ func (d *databaseInfo) checkAndSetDefaults(cf *CLIConf, tc *client.TeleportClien
 		return nil
 	}
 
-	var clusterClient *client.ClusterClient
-	err = client.RetryWithRelogin(cf.Context, tc, func() error {
-		clusterClient, err = tc.ConnectToCluster(cf.Context)
-		return trace.Wrap(err)
-	})
-	if err != nil {
-		return trace.Wrap(err)
-	}
-	defer clusterClient.Close()
+	checker := d.checker
+	if checker == nil {
+		var clusterClient *client.ClusterClient
+		err = client.RetryWithRelogin(cf.Context, tc, func() error {
+			clusterClient, err = tc.ConnectToCluster(cf.Context)
+			return trace.Wrap(err)
+		})
```

Comment:
Nit: Would it make things a little cleaner if all this logic was in a separate helper method? Something like:

```go
func (d *databaseInfo) getChecker(cf *CLIConf, tc *client.TeleportClient) (services.AccessChecker, error) {
  if d.checker != nil {
    return d.checker
  }
  ... // rest of logic to create access checker
}
```